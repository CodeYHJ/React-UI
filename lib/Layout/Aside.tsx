import React, { HtmlHTMLAttributes } from 'react';
import { classPre } from '@lib/utils';
export interface AsideProps extends HtmlHTMLAttributes<HTMLElement> {

}
const c = classPre('layout-aside')

const Aside: React.SFC<AsideProps> = (props) => {
    const { className, ...others } = props
    const cls = [c(), className].filter(Boolean).join(' ')
    return (
        <aside className={cls} {...others}>
            <div className={c('children')}>
                {props.children}

            </div>
        </aside>
    );
}

export default Aside;