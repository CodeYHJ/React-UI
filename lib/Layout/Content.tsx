import React, { HtmlHTMLAttributes } from 'react';
import { classPre } from '@lib/utils';
export interface ContentProps extends HtmlHTMLAttributes<HTMLElement> {

}
const c = classPre('layout-content')

const Content: React.SFC<ContentProps> = (props) => {
    const { className, ...others } = props
    const cls = [c(), className].filter(Boolean).join(' ')
    return (
        <main className={cls} {...others}>
            {props.children}
        </main>
    );
}

export default Content;