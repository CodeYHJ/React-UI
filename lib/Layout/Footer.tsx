import React, { HtmlHTMLAttributes } from 'react';
import { classPre } from '@lib/utils';
export interface FooterProps extends HtmlHTMLAttributes<HTMLElement> {

}
const c = classPre('layout-footer')

const Footer: React.SFC<FooterProps> = (props) => {
    const { className, ...others } = props
    const cls = [c(), className].filter(Boolean).join(' ')
    return (
        <footer className={cls} {...others}>
            {props.children}
        </footer>
    );
}

export default Footer;