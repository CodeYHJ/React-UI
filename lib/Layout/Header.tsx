import React, { HtmlHTMLAttributes } from 'react';
import { classPre } from "@lib/utils"

export interface HeraderProps extends HtmlHTMLAttributes<HTMLElement> {

}
const c = classPre('layout-header')

const Herader: React.SFC<HeraderProps> = (props) => {
    const { className, ...others } = props
    const cls = [c(), className].filter(Boolean).join(' ')
    return (
        <div className={cls} {...others}>
            {props.children}
        </div>
    );
}

export default Herader;