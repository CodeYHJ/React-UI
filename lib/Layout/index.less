.code-ui-layout {
  display: flex;
  flex-grow: 1;
  flex-direction: column;
  &-header {
    margin: 0;
    padding: 0;
  }
  &-content {
    display: flex;
    flex-grow: 1;
    overflow-x: hidden;
  }
  &-footer {
    margin: 0;
    padding: 0;
  }
  &-aside {
    min-width: 200px;
    position: relative;
    &-children {
      height: 100%;
      overflow: hidden;
    }
  }
  &-hasAside {
    display: flex;
    flex-direction: row;
  }
}
