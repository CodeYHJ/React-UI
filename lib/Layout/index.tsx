export { default as Layout } from "./Layout";
export { default as Content } from "./Content";
export { default as Aside } from "./Aside";
export { default as Footer } from "./Footer";
export { default as Header } from "./Header";

