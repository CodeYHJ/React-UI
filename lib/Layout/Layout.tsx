import React, { HtmlHTMLAttributes, ReactElement } from 'react';
import { classPre } from "@lib/utils"
import './index.less'
import Aside from './Aside';
import Content from './Content';
import Footer from './Footer';
type LayoutChildrenType = typeof Aside | typeof Content | typeof Footer | typeof Layout
export interface LayoutProps extends HtmlHTMLAttributes<HTMLElement> {
    children: ReactElement<LayoutChildrenType> | Array<ReactElement<LayoutChildrenType>>
}
const c = classPre('layout')
const Layout: React.SFC<LayoutProps> = (props) => {
    const { className, children, ...others } = props
    let hasAside = false
    if ((children as Array<ReactElement>).length) {
        (children as Array<ReactElement>).forEach(element => {
            if (element.type === Aside) {
                hasAside = true
            }
        })
    }
    const cls = hasAside ? [c(), c('hasAside'), className].filter(Boolean).join(' ') : [c(), className].filter(Boolean).join(' ')

    return (
        <div className={cls} {...others}>
            {props.children}
        </div>
    );
}

export default Layout;