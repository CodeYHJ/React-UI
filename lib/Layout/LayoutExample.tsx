import React from 'react';
import Layout from "./Layout";
import Herader from './Header';
import Content from './Content';
import Footer from './Footer';
import Aside from './Aside';
export interface LayoutExampleProps {

}

const LayoutExample: React.SFC<LayoutExampleProps> = () => {
    return (
        <div>
            <div>
                <h1>第一个例子</h1>
                <Layout style={{ color: '#ffffff' }}>
                    <Herader style={{ background: "#7DBCEA", textAlign: 'center', lineHeight: '50px' }}>Header</Herader>
                    <Content style={{ background: "#108EE9", textAlign: 'center', lineHeight: '50px' }}>Content</Content>
                    <Footer style={{ background: "#7DBCEA", textAlign: 'center', lineHeight: '50px' }}>Footer</Footer>
                </Layout>
            </div>
            <div>
                <h1>第二个例子</h1>
                <Layout style={{ color: '#ffffff' }}>
                    <Herader style={{ background: "#7DBCEA", textAlign: 'center', lineHeight: '50px' }}>Header</Herader>
                    <Layout style={{ color: '#ffffff' }}>
                        <Aside style={{ background: '#3ba0e9', display: 'flex', justifyContent: 'center', alignItems: "center" }}>Aside</Aside>
                        <Content style={{ background: "#108EE9", textAlign: 'center', lineHeight: '50px' }}>Content</Content>
                    </Layout>
                    <Footer style={{ background: "#7DBCEA", textAlign: 'center', lineHeight: '50px' }}>Footer</Footer>
                </Layout>
            </div>
            <div>
                <h1>第三个例子</h1>
                <Layout style={{ color: '#ffffff' }}>
                    <Herader style={{ background: "#7DBCEA", textAlign: 'center', lineHeight: '50px' }}>Header</Herader>
                    <Layout style={{ color: '#ffffff' }}>
                        <Content style={{ background: "#108EE9", textAlign: 'center', lineHeight: '50px' }}>Content</Content>
                        <Aside style={{ background: '#3ba0e9', display: 'flex', justifyContent: 'center', alignItems: "center" }}>Aside</Aside>
                    </Layout>
                    <Footer style={{ background: "#7DBCEA", textAlign: 'center', lineHeight: '50px' }}>Footer</Footer>
                </Layout>
            </div>
            <div>
                <h1>第四个例子</h1>
                <Layout style={{ color: '#ffffff' }}>
                    <Aside style={{ background: '#3ba0e9', display: 'flex', justifyContent: 'center', alignItems: "center" }}>Aside</Aside>
                    <Layout style={{ color: '#ffffff' }}>
                        <Herader style={{ background: "#7DBCEA", textAlign: 'center', lineHeight: '50px' }}>Header</Herader>
                        <Content style={{ background: "#108EE9", textAlign: 'center', lineHeight: '50px' }}>Content</Content>
                        <Footer style={{ background: "#7DBCEA", textAlign: 'center', lineHeight: '50px' }}>Footer</Footer>
                    </Layout>
                </Layout>
            </div>
            <div>
                <h1>第四个例子</h1>
                <Layout style={{ color: '#ffffff' }}>
                    <Aside style={{ background: '#3ba0e9', display: 'flex', justifyContent: 'center', alignItems: "center" }}>Aside</Aside>
                    <Layout style={{ color: '#ffffff' }}>
                        <Herader style={{ background: "#7DBCEA", textAlign: 'center', lineHeight: '50px' }}>Header</Herader>
                        <Content style={{ background: "#108EE9", textAlign: 'center', lineHeight: '50px' }}>Content</Content>
                        <Footer style={{ background: "#7DBCEA", textAlign: 'center', lineHeight: '50px' }}>Footer</Footer>
                    </Layout>
                </Layout>
            </div>
        </div>
    );
}

export default LayoutExample;