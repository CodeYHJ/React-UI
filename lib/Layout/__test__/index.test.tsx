import React from "react"
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Layout from '../index'
const { Content, Header, Footer, Aside } = Layout
describe('Layout', () => {
    describe('render Layout modal', () => {
        it('modale one', () => {
            const json = renderer.create(
                <Layout >
                    <Header  >Header</Header>
                    <Content >Content</Content>
                    <Footer >Footer</Footer>
                </Layout>
            )
            expect(json).toMatchSnapshot()

        })
        it('modale two', () => {
            const json = renderer.create(
                <Layout >
                    <Header  >Header</Header>
                    <Content >
                        <Layout >
                            <Aside>Aside</Aside>
                            <Content >Content</Content>
                        </Layout>
                    </Content>
                    <Footer >Footer</Footer>
                </Layout>
            )
            expect(json).toMatchSnapshot()
        })
        it('modale three', () => {
            const json = renderer.create(
                <Layout >
                    <Header  >Header</Header>
                    <Content >
                        <Layout >
                            <Content >Content</Content>
                            <Aside>Aside</Aside>
                        </Layout>
                    </Content>
                    <Footer >Footer</Footer>
                </Layout>
            )
            expect(json).toMatchSnapshot()
        })
        it('modale four', () => {
            const json = renderer.create(
                <Layout >
                    <Header  >Header</Header>
                    <Content >
                        <Layout >
                            <Header  >Header</Header>
                            <Content >Content</Content>
                            <Footer >Footer</Footer>
                        </Layout>
                    </Content>
                    <Footer >Footer</Footer>
                </Layout>
            )
            expect(json).toMatchSnapshot()
        })
    })

})