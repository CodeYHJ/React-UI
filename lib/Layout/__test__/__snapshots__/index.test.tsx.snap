// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout render Layout modal modale four 1`] = `
<div
  className="code-ui-layout"
>
  <header
    className="code-ui-layout-header"
  >
    Header
  </header>
  <main
    className="code-ui-layout-content"
  >
    <div
      className="code-ui-layout"
    >
      <header
        className="code-ui-layout-header"
      >
        Header
      </header>
      <main
        className="code-ui-layout-content"
      >
        Content
      </main>
      <footer
        className="code-ui-layout-footer"
      >
        Footer
      </footer>
    </div>
  </main>
  <footer
    className="code-ui-layout-footer"
  >
    Footer
  </footer>
</div>
`;

exports[`Layout render Layout modal modale one 1`] = `
<div
  className="code-ui-layout"
>
  <header
    className="code-ui-layout-header"
  >
    Header
  </header>
  <main
    className="code-ui-layout-content"
  >
    Content
  </main>
  <footer
    className="code-ui-layout-footer"
  >
    Footer
  </footer>
</div>
`;

exports[`Layout render Layout modal modale three 1`] = `
<div
  className="code-ui-layout"
>
  <header
    className="code-ui-layout-header"
  >
    Header
  </header>
  <main
    className="code-ui-layout-content"
  >
    <div
      className="code-ui-layout code-ui-layout-hasAside"
    >
      <main
        className="code-ui-layout-content"
      >
        Content
      </main>
      <aside
        className="code-ui-layout-aside"
      >
        <div
          className="code-ui-layout-aside-children"
        >
          Aside
        </div>
      </aside>
    </div>
  </main>
  <footer
    className="code-ui-layout-footer"
  >
    Footer
  </footer>
</div>
`;

exports[`Layout render Layout modal modale two 1`] = `
<div
  className="code-ui-layout"
>
  <header
    className="code-ui-layout-header"
  >
    Header
  </header>
  <main
    className="code-ui-layout-content"
  >
    <div
      className="code-ui-layout code-ui-layout-hasAside"
    >
      <aside
        className="code-ui-layout-aside"
      >
        <div
          className="code-ui-layout-aside-children"
        >
          Aside
        </div>
      </aside>
      <main
        className="code-ui-layout-content"
      >
        Content
      </main>
    </div>
  </main>
  <footer
    className="code-ui-layout-footer"
  >
    Footer
  </footer>
</div>
`;
